balancer-js/src/modules/pools/factory/linear/linear.factory.integration.spec.ts

```diff
@@ -0,0 +1,102 @@
+// yarn test:only ./src/modules/pools/factory/composable-stable/composable-stable.factory.integration.spec.ts
+import { parseFixed } from '@ethersproject/bignumber';
+import { Contract } from '@ethersproject/contracts';
+import { JsonRpcProvider, TransactionReceipt } from '@ethersproject/providers';
+import { expect } from 'chai';
+import dotenv from 'dotenv';
+
+import {
+  LinearCreatePoolParameters,
+  ProtocolId,
+} from '@/modules/pools/factory/types';
+import { BalancerSDK } from '@/modules/sdk.module';
+import { ADDRESSES } from '@/test/lib/constants';
+import { forkSetup } from '@/test/lib/utils';
+import { Network, PoolType } from '@/types';
+import { ERC4626LinearPool, EulerLinearPool } from '@/contracts';
+
+dotenv.config();
+
+const network = Network.MAINNET;
+const rpcUrl = 'http://127.0.0.1:8545';
+const balancer = new BalancerSDK({
+  network,
+  rpcUrl,
+});
+const provider = new JsonRpcProvider(rpcUrl, network);
+const signer = provider.getSigner();
+const addresses = ADDRESSES[network];
+
+describe('creating linear pool', async () => {
+  // Euler pools needs to be created with euler wrapped tokens
+  // const poolType = PoolType.EulerLinear;
+  // const poolTokens = [addresses.DAI, addresses.eDAI];
+  const poolType = PoolType.ERC4626Linear;
+  const poolTokens = [addresses.APE, addresses.sAPE];
+  const linearPoolFactory = balancer.pools.poolFactory.of(poolType);
+  let poolParams: LinearCreatePoolParameters;
+  let signerAddress: string;
+  before(async () => {
+    signerAddress = await signer.getAddress();
+    await forkSetup(
+      signer,
+      [],
+      [],
+      [],
+      `${process.env.ALCHEMY_URL}`,
+      16720000,
+      false
+    );
+    poolParams = {
+      name: 'My-Test-Pool-Name',
+      symbol: 'My-Test-Pool-Symbol',
+      mainToken: poolTokens[0].address,
+      wrappedToken: poolTokens[1].address,
+      upperTarget: '20000',
+      owner: signerAddress,
+      protocolId: ProtocolId.EULER,
```

Comment:
Think we should use a correct protocolId here?

Reply:
Yeah, it was more to keep consitency and make things less confusing. I've updated now.